File: contest/a_tickets/sol/a_tickets.exs
readInt = fn -> IO.read(:stdio, :line) |> String.trim() |> String.to_integer() end

nbbillets = readInt.()
prix1 = readInt.()
prix2 = readInt.()
prix3 = readInt.()

cond do
  nbbillets > 200 ->
    nbbillets = nbbillets - 200
    IO.puts(nbbillets * prix3 + 100 * prix2 + 100 * prix1)

  nbbillets > 100 ->
    nbbillets = nbbillets - 100
    IO.puts(nbbillets * prix2 + 100 * prix1)

  true ->
    IO.puts(nbbillets * prix1)
end

File: contest/a_tickets/sol/a_tickets.py
# Solution by Louis Hasenfratz

nbbillets = int(input())
prix1=int(input())
prix2 = int(input())
prix3 = int(input())
tarif=0
if nbbillets>200:
    nbbillets=nbbillets-200
    print(nbbillets*prix3+100*prix2+100*prix1)
elif nbbillets>100:
    nbbillets = nbbillets - 100
    print(nbbillets * prix2 + 100 * prix1)
else :
    print(nbbillets * prix1)



